### PR TITLE
Fix typescript errors #653

### LINKED ISSUE #653: Fix typescript errors in test files
Run yarn tsc

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

Run yarn tsc and fix all of the typescript warnings 

## Notes 

The catalogue item property type is incorrect. it currently value | undefined when it should be value | null. This doesn't cause any bugs and will be addressed in the react hook form prs  

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

closes #653

### CODE REVIEW

File: src/api/catalogueCategories.test.tsx [hunk 1]
```diff
@@ -681,9 +697,10 @@ describe('catalogue categories api functions', () => {
   });
   describe('useAddCatalogueCategoryProperty', () => {
     let mockDataAddProperty: AddPropertyMigration;
+    getCatalogueCategoryById('4');
```
Comment (joelvdavies) [inline src/api/catalogueCategories.test.tsx:700]:
```suggestion
```
What is this doing?

File: src/catalogue/items/deleteCatalogueItemDialog.component.test.tsx [hunk 2]
```diff
@@ -35,15 +38,28 @@ describe('delete Catalogue Category dialogue', () => {
...
+      ...CREATED_MODIFIED_TIME_VALUES,
     };
     props = {
       open: true,
       onClose: onClose,
       catalogueItem: catalogueItem,
       onChangeCatalogueItem: onChangeCatalogueItem,
     };
-    user = userEvent; // Assigning userEvent to 'user'
+    user = userEvent.setup(); // Assigning userEvent to 'user'
```
Comment (joelvdavies) [inline src/catalogue/items/deleteCatalogueItemDialog.component.test.tsx:62]:
```suggestion
    user = userEvent.setup();
```

**Review by joelvdavies — COMMENTED** [review]
Two minor comments.

File: src/api/catalogueCategories.test.tsx (hunk 1 above)
Comment (joshuadkitenge) [reply src/api/catalogueCategories.test.tsx:700]: Good spot I forgot to remove it

**Review by joelvdavies — APPROVED** [review]

**Pull request merged**